### PR TITLE
test: suite improvements

### PR DESCRIPTION
**Pull request by 0xAustrian**
Closes BAL-57, BAL-58

### CODE REVIEW

**Review by 0xJabberwock — CHANGES_REQUESTED** [review]
Very nice job! 💯 

Let's fix the issues raised by the linter, particularly the _unused variable_ warning.

File: foundry.toml [hunk 1]
```diff
@@ -27,8 +27,8 @@ out = 'out-via-ir'
 src = 'src/interfaces/'
 
 [fuzz]
-runs = 800
-max_test_rejects = 2_000_000
+runs = 1000
+max_test_rejects = 2_500_000
```
[the remark this answers is not in the document]
Comment (0xAustrian) [reply foundry.toml:31]: I tried but some cases still needed that amount. And since I increased the runs the max rejects had to also increase.

**Comment by 0xAustrian** [comment]
Fixed most of the linter warnings. Expect for `673:5   warning  Variable "_tokensLength" is unused  no-unused-vars` which seems like a bug in solhint. The variable is being used inside the modifier.

File: test/unit/BPool.t.sol [hunk 2]
```diff
@@ -157,9 +165,9 @@ abstract contract BasePoolTest is Test, BConst, Utils, BMath {
 
   function _assumeCalcOutGivenIn(
     uint256 _tokenInBalance,
-    uint256 _tokenInDenorm,
-    uint256 _tokenOutBalance,
-    uint256 _tokenOutDenorm,
+    uint256,
+    uint256,
+    uint256,
```
[the remark this answers is not in the document]
Comment (0xJabberwock) [reply test/unit/BPool.t.sol:182]: Why to keep them as inputs then?

**Comment by 0xJabberwock** [comment]
Fixing all the linter warnings, specially those related to a function position refactor, introduced many new changes to the PR's diff that make it difficult to follow it, worsening its auditability—as a result, a rollback has been applied.

**Review by 0xJabberwock — APPROVED** [review]
🚀

**Pull request merged**